src/lib.rs

```diff
@@ -826,16 +823,16 @@ impl<K: EnrKey> rlp::Decodable for Enr<K> {
         let mut content = BTreeMap::new();
         let mut prev: Option<String> = None;
         for _ in 0..decoded_list.len() / 2 {
-            let key = decoded_list.remove(0);
-            let value = decoded_list.remove(0);
+            let key = decoded_list.remove(0).data()?;
+            let value = decoded_list.remove(0).data()?;
```

Comment:
My concern was this being a recursive list, we strip off the rlp header here. But when interpreting a specific field, I guess we can always account for that.